src/components/Filters/CheckboxFilter.tsx

```diff
@@ -0,0 +1,58 @@
+import { BaseFilter } from "src/components/Filters/BaseFilter";
+import { Filter } from "src/components/Filters/types";
+import { Checkbox } from "src/inputs";
+import { TestIds } from "src/utils/useTestIds";
+
+export type CheckboxFilterProps<V> = {
+  label?: string;
+  onValue?: V | undefined;
+  offValue?: V | undefined;
+  defaultValue?: V | undefined;
+};
+
+/**
+ * Provides a two-state "on/off" filter.
+ *
+ * By default the on/off values are `on === true` and `off === undefined`.
+ *
+ * You can flip the on/off values by passing `onValue: false`, in which case
+ * `on === false` and off === undefined`.
+ *
+ * Or you can set on/off directly, by passing both `onValue` and `offValue`, even to
+ * non-boolean values, i.e. `onValue: "foo", offValue: "bar"`.
+ */
+export function checkboxFilter<V = boolean>(props: CheckboxFilterProps<V>): (key: string) => Filter<V> {
+  return (key) =>
+    new CheckboxFilter(key, {
+      // If the user has set the offValue, that should be the default b/c we're only a two-state
+      defaultValue: props.offValue,
+      ...props,
+    });
+}
+
+class CheckboxFilter<V> extends BaseFilter<V, CheckboxFilterProps<V>> implements Filter<V> {
+  render(
+    value: V | undefined,
+    setValue: (value: V | undefined) => void,
+    tid: TestIds,
+    inModal: boolean,
+    vertical: boolean,
+  ): JSX.Element {
+    const { defaultValue, onValue = true as any as V, offValue = undefined, ...props } = this.props;
+    return (
+      <Checkbox
+        {...props}
+        selected={value === undefined ? false : value === onValue}
+        label={this.label}
+        onChange={(on) => {
+          setValue(on ? onValue : offValue);
+        }}
+        {...this.testId(tid)}
+      />
+    );
+  }
+
+  get hideLabelInModal() {
```

Comment:
Ah nice, I forgot we already had a way to hide labels in the modal; that worked out...

Reply:
Sounds good, thanks for the feedback. I will mention it to the team on Monday. I wanted to take this ticket for Hackday because I noticed design moving to this new pattern of checkbox filters and we've been using the toggle instead because this hadn't been built yet.

Reply:
Cool! Yeah, cleaning up/adding to Beam is great hack day work; sorry, I got on my usual "ux consistency" soapbox, so didn't mean for that to be demotivating.

Since this is something design consistently wants (i.e. hopefully is the new pattern), and sounds like it just never made it "to the top of a squad's feature backlog" (which is common for small-ish Beam updates), then implementing these updates on hack days is a perfect use of the carved-out hack day time imo.

...if you wanted to see if design was cool with just removing the old toggle filter, and just whole-sale cut over to the new checkbox filter, that would also be amazing, but happy to leave that to later/follow up discussions. Really not a big deal to leave the old one as-is.